### PR TITLE
Helix test: remove hack to copy currently built runtime.

### PR DESCRIPTION
**Pull request by rahku**
remove hack to copy currently built runtime.
Instead add dependency to currently built runtime package

### CODE REVIEW

**Comment by rahku** [comment]
PTAL @ramarag @gkhanna79

File: tests/src/Common/test_runtime/test_runtime.csproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (ramarag) [inline tests/src/Common/test_runtime/test_runtime.csproj]: What does this project do ? Can you add Comments ?

**Comment by rahku** [comment]
Will do. All the test projects need to add dependency to currently built runtime as they require that to run. In addition the version number of built runtime can change so all project.json needs to be dynamically generated. In order to avoid regenerating project.json for all the tests projects...I added a common project for all the tests. This project is adding common dependencies to all the tests like dependency on Microsoft.netcore.runtime.coreclr ..etc.  Project.json for this project is dynamically generated  via task CreateTestRuntimeJsonFile in runtest.proj.

File: tests/runtest.proj [hunk 2]
```diff
@@ -315,7 +315,46 @@ namespace $([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").
...
+
+  </Target>
+
   <Target Name="Build">
+
+    <MSBuild Projects="$(MSBuildProjectFile)"
+             Targets="CreateTestRuntimeJsonFile"
+             Condition=" '$(NoBuild)'!='true' " />
+
+    <MSBuild Projects="src\Common\test_runtime\test_runtime.csproj"
```
Comment (ramarag) [inline tests/runtest.proj]: A comment here would be nice

File: tests/src/Common/test_runtime/test_runtime.csproj [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <OutputType>Library</OutputType>
```
Comment (ramarag) [inline tests/src/Common/test_runtime/test_runtime.csproj]: does it need to have a OutputType ?

File: tests/runtest.proj (hunk 2 above)
Comment (rahku) [inline tests/runtest.proj]: I will distribute the explaination I gave above to right places.

File: tests/src/Common/test_runtime/test_runtime.csproj (hunk 3 above)
Comment (rahku) [inline tests/src/Common/test_runtime/test_runtime.csproj]: don't care about the output.....all that is needed is project.lock.json

Comment (ramarag) [inline tests/src/Common/test_runtime/test_runtime.csproj]: Unless msbuild complaints, please remove it

Comment (rahku) [inline tests/src/Common/test_runtime/test_runtime.csproj]: It does

Comment (rahku) [inline tests/src/Common/test_runtime/test_runtime.csproj]: It tries to create an exe and require file with main method

**Comment by ramarag** [comment]
LGTM minus the comments

**Pull request merged**